### ISSUE TITLE
Main window size increases each time VS Code is opened

### ISSUE THREAD

**Issue by tulipan-it** [microsoft/vscode#170001]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.74.2
- OS Version: Windows 10.0.19044 Build 19044

Steps to Reproduce:

1. Put VS Code in a non-maximized size mode.
2. Resize the main window so that its right and bottom border is near the right and bottom edge of the screen. The top and left edge should have some space between the screen's edges.
3. Repeatedly open and close the app.
4. Watch the window slowly increase its size towards the right and bottom edge.
5. Watch it grow even outside of the screen.

**Comment by bpasero** [comment]
Some questions around the setup:
* is this multi monitor setup
* do you configure scaling or zooming on Windows for the screen?
* does it reproduce without scaling or zooming

**Comment by Chaos-070799** [comment]
It reproduces I'm assuming to my google calendar....  what happens I was locked out of my google act!  Being states away from home I needed my google act back I had just started my own business and they kept asking for passwords every little bit...  to the point I forgot my password!  I reached out to Google they had me reach out to my significant other with access to prove my identity and get me back in my act!  Since then my phone has cut up but it's cool, on oct ? I texted in calendar help me get my act back and he responded....  since then, this is been going on

**Comment by Chaos-070799** [comment]
And no, it appears that I only have access to his phone he does not have access to mine! Total

**Comment by tulipan-it** [comment]
No multi monitor. It's a standalone 17'' laptop (1920 x 1080), pretty much out of the box with no scaling or zooming configured.
![Screenshot (71)](https://user-images.githubusercontent.com/52129590/209633527-7abb20d8-f713-4d2d-9cc3-afbcd6e44cac.png)

**Comment by bpasero** [comment]
Can you try to reproduce with our nightly insider builds? You can give our preview releases a try from: https://code.visualstudio.com/insiders/

**Comment by tulipan-it** [comment]
Yes, it reproduces. I downloaded and installed and run directly with no extra configuration made. Here's screenshots from before, and after 10 openings. I just clicked the upper right cross and then doubleclicked the desktop icon, 10 times.

Before:

![VSCode before](https://user-images.githubusercontent.com/52129590/209660395-8577fbe5-193a-4a3e-b132-4b313122534f.png)

After 10 re-openings:

![VSCode after 10 openings](https://user-images.githubusercontent.com/52129590/209660434-dc9d0e45-4489-4055-9c58-f2233924bb1b.png)

**Comment by tulipan-it** [comment]
Just to make sure that I didn't fumble when clicking the cross, I repeated by closing using Alt+F4, with the same result.

**Comment by tulipan-it** [comment]
Btw, the same happens on Linux, but with the difference that the drifting stops when the right and bottom edges of the screen are reached. Here's the screenshot from my Linux distro Lubuntu, run in VirtualBox, after some re-openings of VS Code:

![screen](https://user-images.githubusercontent.com/52129590/209663341-c16ef0a3-4f75-419b-ab45-a931d81bdb86.jpg)

**Comment by bpasero** [comment]
Does it reproduce when you configure `window.titleBarStyle: native`?

**Comment by tulipan-it** [comment]
Yes, unfortunately.

![Screenshot (74)](https://user-images.githubusercontent.com/52129590/209711757-5ee0fdf9-b2b5-45c1-83c9-d9f32537d0ef.png)

**Comment by bpasero** [comment]
Since I was not able to reproduce so far, maybe you could record a Gif/Video how you start and how it ends, then I will try to reproduce exactly like that. For example https://gifcap.dev/ allows to record the screen as Gif and then you can attach here.

**Comment by tulipan-it** [comment]
Ok, I've made a recording. Btw, I tested with other apps, like Notepad++, and it doesn't happen with them. Only VS Code.

![Recording 2022-12-28 at 09 24 38](https://user-images.githubusercontent.com/52129590/209782584-3c0b9b15-e209-4512-9e2a-99dc8631e226.gif)

**Comment by bpasero** [comment]
When you run `code-insiders --verbose` we produce verbose logs that include information for window state (size, position). Can you run the steps from the gif attaching the logs you see from the console? For example, the line that has `window#validateWindowState` is relevant, as well as `[WindowsStateHandler] onBeforeShutdown`

**Comment by tulipan-it** [comment]
I couldn't get the gif tool to work in this scenario. It took too long time and the recording kept abort with an error. However, I took screenshots instead of the relevant parts. I opened and closed two times, took 4 screenshots of each. The first 3 in each set is from the beginning of the log file, after VS Code has been opened. You can find "window#validateWindowState" on #3. The last, #4, is from after I have closed it, and contains "onBeforeShutdown". Here's the first set.

![vscode-first-1](https://user-images.githubusercontent.com/52129590/209987583-8261aba2-1413-4f3f-adff-318d985663eb.png)

![vscode-first-2](https://user-images.githubusercontent.com/52129590/209987610-b9a5b3a0-24c0-4dbe-9f10-8e0b22ce68bc.png)

![vscode-first-3](https://user-images.githubusercontent.com/52129590/209987626-d4b801b0-cae1-4dc2-9043-1e874ad3cece.png)

![vscode-first-4](https://user-images.githubusercontent.com/52129590/209987653-a16ce814-14a4-45a5-a083-a67ae877e15c.png)

**Comment by tulipan-it** [comment]
Here's the second:

![vscode-second-1](https://user-images.githubusercontent.com/52129590/209987728-e91afeed-de4f-4e17-b0c8-245a93edffef.png)

![vscode-second-2](https://user-images.githubusercontent.com/52129590/209987753-5c141007-97a3-4fed-b6db-2380e3c6273d.png)

![vscode-second-3](https://user-images.githubusercontent.com/52129590/209987772-d076ca8d-397d-4974-8bcf-68e272f5d49e.png)

![vscode-second-4](https://user-images.githubusercontent.com/52129590/209987788-6ef4e1dc-e303-406a-b253-9c40408623df.png)

**Comment by bpasero** [comment]
@tulipan-it thanks this is useful, here is the relevant log extracted and notice how the position but mainly the size of the window seems to change between the restarts:

<img width="764" alt="image" src="https://user-images.githubusercontent.com/900690/209990649-9b7c86cd-d435-454f-9851-f11b8aa24a40.png">

<img width="248" alt="image" src="https://user-images.githubusercontent.com/900690/209990671-a95e71d3-171a-4adc-b232-c40372a684e1.png">

<img width="541" alt="image" src="https://user-images.githubusercontent.com/900690/209990711-0b3042ec-3dd3-4017-8b19-d8af6b4a3cbd.png">

<img width="241" alt="image" src="https://user-images.githubusercontent.com/900690/209990724-bde330f3-105e-4d88-9424-f157e17f3fd6.png">

In all these cases I would have expected `x`, `y`, `width` and `height` to stay the same unless you as a user would resize or move the window.

Is there any special window manager or 3rd party software installed?

**Comment by tulipan-it** [comment]
No, I try to keep it as clean as possible regarding such tools. The only thing I recall is Armoury Crate, which is an optional, advanced low level system manager from ASUS. However, it mostly concerns colors, CPU and memory stuff.

Notice that it also happens on a virtual machine running Linux. And that it doesn't happen for other apps but VS Code. So I draw the conclusion that:

1. It is VS Code specific
2. It has something to do with the physical machine, not the OS

It's an 17'' ASUS laptop. Maybe 17'' laptops are a bit uncommon. If you or your team has access to such, it might be interesting to test on that.

Otherwise, this is not really a big deal. I just wanted to report it. I do most of my development on Linux and there the drifting at least stay within the screen and bottom taskbar. If you cannot reproduce and no other reports of this are coming in, I think it must have a low priority.

**Comment by bpasero** [comment]
Got it, esp. the part about this reproducing on Linux too is very strange. 

Curious, does it also happen when the window is maximized?

**Comment by tulipan-it** [comment]
Nope.

**Comment by bpasero** [comment]
Oh, just found this upstream issue from the UI framework we use: https://github.com/electron/electron/issues/27651

I bet its your issue as well, sounds very much similar.

**Comment by tulipan-it** [comment]
It's probably related somehow. But they talk about multiple screens and different scaling. I have only one and keep it at 100%.

**Comment by bpasero** [comment]
Marking as upstream: we get different window bounds after setting bounds, so the window moves on the screen. I am not aware of a way for us to fix this and https://github.com/electron/electron/issues/27651 sounds very much related.

**Comment by jeffpapp** [comment]
Not sure this is related to the same issue but I notice that when I open a second instance of VS Code on Windows it doesn't size the same as the first.

For example:

1. Open VS Code and maximize the window and then close it
2. Reopen VS Code and the first instance will be maximized like I would expect since the last instance closed was maximized.
3. Open a second instance of VS Code and this one will never be maximized.  I would expect it to be the same size as the other VS Code window already open

**Comment by bpasero** [comment]
It is unrelated, that is configurable with `window.newWindowDimensions` setting.
